apps/fyle/tasks.py

```diff
@@ -104,8 +105,16 @@ def create_expense_groups(workspace_id: int, configuration: Configuration, fund_
 
             expense_objects = Expense.create_expense_objects(expenses)
 
+            expenses_object_ids = [expense_object.id for expense_object in expense_objects]
+
+            expense_filters = ExpenseFilter.objects.filter(workspace_id=workspace_id)
+            if expense_filters:
+                final_query = construct_expense_filter_query(expense_filters)
+                Expense.objects.filter(final_query, id__in=expenses_object_ids, expensegroup__isnull=True, org_id=workspace.fyle_org_id).update(is_skipped=True)
+
+            filtered_expenses = Expense.objects.filter(is_skipped=False, id__in=expenses_object_ids, expensegroup__isnull=True, org_id=workspace.fyle_org_id)
```

Comment:
can we just assign the value of `expense_objects` to `filtered_expenses` in the else block of 111th line? this query seems unnecessary, please also think of all edge cases

Reply:
if you do this, in the if block, assign value to filtered expense

Reply:
Something like this ? 
```suggestion
            expense_objects = Expense.create_expense_objects(expenses)
            expense_filters = ExpenseFilter.objects.filter(workspace_id=workspace_id)

            if expense_filters:
                expenses_object_ids = [expense_object.id for expense_object in expense_objects]
                final_query = construct_expense_filter_query(expense_filters)
                Expense.objects.filter(final_query, id__in=expenses_object_ids, expensegroup__isnull=True, org_id=workspace.fyle_org_id).update(is_skipped=True)
                filtered_expenses = Expense.objects.filter(is_skipped=False, id__in=expenses_object_ids, expensegroup__isnull=True, org_id=workspace.fyle_org_id)
                
                ExpenseGroup.create_expense_groups_by_report_id_fund_source(
                    filtered_expenses, configuration, workspace_id
                )
            else:
                ExpenseGroup.create_expense_groups_by_report_id_fund_source(
                    expense_objects, configuration, workspace_id
                )
```

Reply:
yes. `ExpenseGroup.create_expense_groups_by_report_id_fund_source()` can keep this out of block. In the else block you can just assign filtered_expenses = expense_objects